### PR TITLE
Bug 1723422 - properly handle pagination url

### PR DESCRIPTION
**Pull request by jmrodri**
* set path and query string properly
* fix getNextImageURL parsing of path url
* add unit test for getNextImageURL

### CODE REVIEW

File: registries/adapters/oauth/client.go [hunk 1]
```diff
@@ -81,7 +81,16 @@ func (c *Client) NewRequest(path string) (*http.Request, error) {
 	if err != nil {
 		return nil, err
 	}
-	req.URL.Path = path
+
+	// path might be a fully qualified URL and we only want the path
+	// and query bits
+	thebits, err := url.Parse(path)
```
Comment (shawn-hurley) [inline registries/adapters/oauth/client.go]: personally, I would change the variable name but not a blocker

File: registries/adapters/oauth/client.go [hunk 2]
```diff
@@ -81,7 +81,16 @@ func (c *Client) NewRequest(path string) (*http.Request, error) {
...
+	if err != nil {
+		return nil, err
+	}
+
+	req.URL.Path = thebits.Path
```
Comment (shawn-hurley) [inline registries/adapters/oauth/client.go]: could we just set `req.URL = thebits`?

**Review by shawn-hurley — APPROVED** [review]
I think it will fix the problem, just some nits.

File: registries/adapters/oauth/client.go (hunk 2 above)
Comment (jmrodri) [reply registries/adapters/oauth/client.go]: The comment on `NewRequest` states they use the base URL from the client. So I wanted to keep the same functionality. 
```golang
// NewRequest - creates and returns a *http.Request assuming the GET method.
// The base URL configured on the Client gets used with its Path component
// replaced by the path argument. If a token is available, it is added to the
// request automatically. "Accept: application/json" is added to all requests.
// The caller should customize the request as necessary before using it.
```

File: registries/adapters/oauth/client.go (hunk 1 above)
Comment (jmrodri) [reply registries/adapters/oauth/client.go]: changed to `pathAsUrl` let me know if that is not okay.

Comment (jmrodri) [reply registries/adapters/oauth/client.go]: actually changed it to `pathAsURL` since `pathAsUrl` pissed off lint :)

**Pull request merged**